Fixed #1707. add user tagging to review bookmarks

bobi7 opened the pull request:
ref #1707

Commit:
Fixed #1704 adds tagging users to review bookmarks comment feature.

Commit:
added checks for when tagged users isnt set. backwards compatability

Commit:
added in button to filter bookmarks by what user has been tagged in

bobi7 commented:
Notes on stuff to fix:

- [x] remove the word "description" and "Created At". put all the review bookmarks info on the same line

- [x] in the notes view (when you select a bookmark) show the tagged users in the comment

- [x] make the comments section look more like a chat ie. dont need the wide div

- [x] add tooltips to the buttons under bookmark comments like the go to map button in top right

- [x] dont create more than 1 bookmark for the same feature/item. If the user tries to publish a bookmark but it already exists, the modal should show the chat logs and allow the user to write a comment

- [x]  clear map when user selects to nagivate to bookmarked review from review bookmarks panel

brianhatchl commented:
I found that if a conflation review is already active in the map, navigating to a bookmarked review from the Review Bookmarks panel creates duplicates of the conflation review controls in the map.

Commit:
multiple improvements added to review bookmarks

Commit:
when launching view review bookmark on map, clear map first

Commit:
dont create more than 1 bookmark for the same feature/item

brianhatchl committed:
Merge remote-tracking branch 'origin/hoot2x' into 1707

# Conflicts:
#	css/hoot/modules/_global.scss

Commit:
Merge branch 'hoot2x' into 1707

brianhatchl committed:
Merge remote-tracking branch 'origin/hoot2x' into 1707

brianhatchl committed:
check for requested review bookmark before requesting new review

when loading a merged layer

Commit:
fixed issue when reviewing feature that had the same feature below it

Commit:
added taking to publish bookmark dialog box

brianhatchl committed:
simplifying this

but we really shouldn't reload the review layer if it's already in the
map but that will have to be in a separate ticket

brianhatchl reviewed (approved):
:turtle: LGTM, added #1791 for future improvement
